fix(ci): version-tag workflow fails because git describe picks up non-semver tags

## Bug

The version-tag workflow fails with:

```
Run NEW_TAG="vstable..1"
fatal: 'vstable..1' is not a valid tag name.
Error: Process completed with exit code 128.
```

## Root Cause

`git describe --tags --abbrev=0` returns the most recent reachable tag regardless of name. The `stable` git tag (which should always track the latest release) sits ahead of `v4.0.1` on `main`, so `git describe` returns `stable` instead of `v4.0.1`.

The semver parser then tries to split `stable` by `.`, producing `MAJOR=stable`, `MINOR=""`, `PATCH=""`, which bumps to `vstable..1` — an invalid git tag name.

## Fix

1. Add `--match 'v[0-9]*'` to the `git describe` command so only semver tags are considered
2. Add a validation guard after semver parsing to fail fast with a clear error if parsing produces non-numeric values
3. Add a step to move the `stable` git tag to the new release commit on each version bump, keeping it always pointing to the latest release

--- a/.github/workflows/version-tag.yml
+++ b/.github/workflows/version-tag.yml
@@ -35,8 +35,9 @@ jobs:
       - name: Get last version tag
         id: last_tag
         run: |
-          # Get the last tag, or default to v0.0.0 if no tags exist
-          LAST_TAG=$(git describe --tags --abbrev=0 2>/dev/null || echo "v0.0.0")
+          # Get the last semver tag, or default to v0.0.0 if no tags exist
+          # --match 'v[0-9]*' filters out non-semver tags (e.g. 'stable', 'backup-*')
+          LAST_TAG=$(git describe --tags --abbrev=0 --match 'v[0-9]*' 2>/dev/null || echo "v0.0.0")
           echo "tag=${LAST_TAG}" >> $GITHUB_OUTPUT
           echo "Last tag: ${LAST_TAG}"
 
@@ -105,6 +106,12 @@ jobs:
           
           # Parse semantic version
           IFS='.' read -r MAJOR MINOR PATCH <<< "$VERSION"
+          
+          # Validate parsed version components are numeric
+          if ! [[ "$MAJOR" =~ ^[0-9]+$ && "$MINOR" =~ ^[0-9]+$ && "$PATCH" =~ ^[0-9]+$ ]]; then
+            echo "::error::Failed to parse semver from tag '${LAST_TAG}': MAJOR=${MAJOR}, MINOR=${MINOR}, PATCH=${PATCH}"
+            exit 1
+          fi
           
           # Bump the appropriate component
           case "$BUMP_TYPE" in
@@ -226,6 +233,16 @@ jobs:
           git push origin "${NEW_TAG}"
           
           echo "Pushed commit and tag ${NEW_TAG}"
+
+      - name: Move stable tag to new release
+        run: |
+          NEW_TAG="${{ steps.new_version.outputs.tag }}"
+          
+          # Move the stable tag to point to this release commit
+          git tag -fa stable -m "Latest stable release: ${NEW_TAG}"
+          git push origin stable --force
+          
+          echo "Moved stable tag to ${NEW_TAG}"
 
       - name: Create GitHub release
         run: |

